backends/files.go

```diff
@@ -349,20 +319,37 @@ func (o *Files) CheckAcl(username, topic, clientid string, acc int32) (bool, err
 
 	fileUser, ok := o.Users[username]
 
-	//If user exists, check against his acls and common ones. If not, check against common acls only.
+	// If user exists, check against his acls and common ones. If not, check against common acls only.
+	// But first check if the topic was explicitly denied and refuse to authorize if so.
 	if ok {
 		for _, aclRecord := range fileUser.AclRecords {
-			if TopicsMatch(aclRecord.Topic, topic) && (acc == int32(aclRecord.Acc) || int32(aclRecord.Acc) == MOSQ_ACL_READWRITE || (acc == MOSQ_ACL_SUBSCRIBE && topic != "#" && (int32(aclRecord.Acc) == MOSQ_ACL_READ || int32(aclRecord.Acc) == MOSQ_ACL_SUBSCRIBE))) {
-				return true, nil
+			match := TopicsMatch(aclRecord.Topic, topic)
+			if match {
+				if aclRecord.Acc == MOSQ_ACL_DENY {
+					return false, nil
+				}
+
+				if acc == int32(aclRecord.Acc) || int32(aclRecord.Acc) == MOSQ_ACL_READWRITE || (acc == MOSQ_ACL_SUBSCRIBE && topic != "#" && (int32(aclRecord.Acc) == MOSQ_ACL_READ || int32(aclRecord.Acc) == MOSQ_ACL_SUBSCRIBE)) {
+					return true, nil
```

Comment:
Shouldn't we process ann DENY before returning true ?

I think we should even process deny from both topic & pattern (`fileUser.AclRecords` and `o.AclRecords`) before any access, shouldn't we ?

Reply:
Isn't this exactly that? https://github.com/iegomez/mosquitto-go-auth/pull/157/files#diff-e804824ad43abd98f9af0bed229f2641cc6f240856a1ff20d17e2599a4ba4e49R328

Reply:
Oooohhh, I just realized what you were saying and you were totally right. Sorry for that lapsus.